pkg/nsx/services/vpc/builder.go

```diff
@@ -65,9 +66,9 @@ func buildNSXVPC(obj *v1alpha1.NetworkInfo, nsObj *v1.Namespace, nc common.VPCNe
 
 func buildNSXLBS(obj *v1alpha1.NetworkInfo, nsObj *v1.Namespace, cluster, lbsSize, vpcPath string, relaxScaleValidation *bool) (*model.LBService, error) {
 	lbs := &model.LBService{}
-	lbsName := util.GenerateIDByObjectByLimit(obj, common.MaxNameLength)
```

Comment:
Please also update LBStore's keyFunc. If we use "default" as the Id for the lbs created in any VPC, we can't use it as the key since it is not unique any more. 
Besides, you may also need to check the consumption logic which now uses the vpc.Id (previously the same as lbs's id) as the key to get lbs in a VPC.

I would prefer to add these two changes in this patch,
1. Use "VPC.ID-lbs.ID" as the key in LBStore,
2. Introduce a new public function in LBStore to get the default LBS by VPC, then constructs the lbs Key as "VPC.ID-default", and call GetByKey to find the lbs. As for the origianl caller with GetByKey(VPC.id), update to consume the new function.

Reply:
Agree with wenying.

Reply:
If one VPC has only one LBS, it should be OK. Not sure if we can guarantee this.

Reply:
If one VPC has only one LBS, it should be OK. Not sure if we can guarantee this.
@TaoZou1 @dantingl Any suggestion?

Reply:
currently, one VPC has only one LBS